tests/conftest.py

```diff
@@ -55,6 +45,111 @@ def pytest_ignore_collect(path, config):
             return True
 
 
+def pytest_configure(config):
+    config.addinivalue_line(
+        'markers',
+        'bdb(): use bigchaindb')
+
+
+@pytest.fixture(autouse=True)
+def _bdb_marker(request):
+    if request.keywords.get('bdb', None):
+        request.getfixturevalue('_bdb')
+
+
+@pytest.fixture(autouse=True)
+def _restore_config(_configure_bigchaindb):
+    from bigchaindb import config, config_utils
+    config_before_test = copy.deepcopy(config)
+    yield
+    config_utils.set_config(config_before_test)
+
+
+@pytest.fixture
+def _restore_dbs(request):
```

Comment:
Good observation! It is not. Perhaps we can just remove it.

It was initially an `autouse` fixture to drop databases that would have potentially been created in a test, and not dropped after. But I had omitted to take into account the databases that may potentially be created in a different process when `pytest-xdist` is used. Some code was added to take that into account, but I started thinking that this required perhaps too many assumptions. So I was undecided and removed the `autouse` option but left the fixture there.

From my understanding, (with rethinkdb), there are three databases before tests are run:

* the bigchaindb db (e.g.: `bigchain`, available via the config)
* `rethinkdb`
* `test`

during tests, there is one test database per session: 

* `bigchain_test` if only one test run is executed at a time

OR, if `pytest-xdist` is used, e.g.:

* `bigchain_test_gw0`
* `bigchain_test_gw1`
* `bigchain_test_gw2`
* `bigchain_test_gw3`

if tests are executed on a four core machine.

So a cleanup fixture could delete all other databases on the basis of the assumption that the other databases were created during some test(s), and not dropped afterwards.

But when `pytest-xdist` the database system is shared by each process, so a process could drop a database that is being used by another process.